From 03dc33028f375035adadbcc93fecfae30717ea2b Mon Sep 17 00:00:00 2001
From: Jeff Law <jlaw@ventanamicro.com>
Date: Fri, 25 Aug 2023 16:23:06 -0600
Subject: [PATCH 13/30] [committed] RISC-V: Fix minor testsuite problem with
 zicond

I thought I had already fixed this, but clearly if I did, I didn't include it
in any upstream commits.

With -Og the optimizers are hindered in various ways and this prevents using
zicond.  So skip this test with -Og (it was already being skipped at -O0).

gcc/testsuite
	* gcc.target/riscv/zicond-primitiveSemantics.c: Disable for -Og.
---
 gcc/testsuite/gcc.target/riscv/zicond-primitiveSemantics.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/testsuite/gcc.target/riscv/zicond-primitiveSemantics.c b/gcc/testsuite/gcc.target/riscv/zicond-primitiveSemantics.c
index 76c5019a992..47d4e4c5683 100644
--- a/gcc/testsuite/gcc.target/riscv/zicond-primitiveSemantics.c
+++ b/gcc/testsuite/gcc.target/riscv/zicond-primitiveSemantics.c
@@ -1,7 +1,7 @@
 /* { dg-do compile } */
 /* { dg-options "-march=rv64gc_zicond -mabi=lp64d" { target { rv64 } } } */
 /* { dg-options "-march=rv32gc_zicond -mabi=ilp32f" { target { rv32 } } } */
-/* { dg-skip-if "" { *-*-* } {"-O0"} } */
+/* { dg-skip-if "" { *-*-* } {"-O0" "-Og"} } */
 
 long primitiveSemantics_00(long a, long b) { return a == 0 ? 0 : b; }
 
-- 
2.25.1

